### ISSUE TITLE
add restart command pls

### ISSUE THREAD

**Issue by ramanveerji** [Samfun75/BookdlBot#7]
the bot becomes unresponsive after some time and needs to be restarted to fix. It will be a nice idea to add restart command in bot itself so that restarting becomes easy from telegram itself..

**Comment by Samfun75** [comment]
Sorry for late reply, I have been busy.

And for the request you asked, I think its better to find out why it is getting stuck in the first place and fix it instead of a temporary fix which is restarting the bot... in short if the bot is stuck, it will not be able to accept commands.

Can you provide these details please? since this never happened on my machine (*typical programmer excuse*)

- What was the bot doing when it got stuck? (Normal downloading or convert and download)
- Is the bot on heroku or other server?
- Can you provide relevant logs?

**Comment by ramanveerji** [comment]
The bot is on heroku. If bot is not used for some time say a day or two then it becomes unresponsive and needs to be restarted. Seems like some pyrogram issue. Adding a restart button will definitely help..

**Comment by Samfun75** [comment]
It is not a pyrogram issue, it might be an issue with libgen blocking the heroku dyno IP and restarting changes the IP. Can you check if the search from downloaded is working when the bot doesn't responds to search query?

I'll investigate more when I have time, but in the mean time if the bot responds to search from the downloaded and not to the normal search, It might be an issue with block IP so I'll try adding another branch with restart command

**Comment by ramanveerji** [comment]
Yes, downloaded search works i guess but normal is not. Adding restart command will be helpful.

**Comment by Samfun75** [comment]
I have looked at this issue again and there is no clean method to restart heroku dyno from the app code it self.

The only option I came across is crashing the app and heroku starts it up again. but if this is done frequently within 10 minutes (might change in the future) heroku might put it to timeout [Source](https://devcenter.heroku.com/articles/dynos#dyno-crash-restart-policy)

Since crashing the app isn't ideal, I'll be closing this issue.
